server/src/main/java/org/opensearch/repositories/IndexId.java

```diff
@@ -51,23 +53,36 @@
  */
 @PublicApi(since = "1.0.0")
 public final class IndexId implements Writeable, ToXContentObject {
-    protected static final String NAME = "name";
-    protected static final String ID = "id";
+    static final String NAME = "name";
+    static final String ID = "id";
+    static final String SHARD_PATH_TYPE = "shard_path_type";
+    public static final int DEFAULT_SHARD_PATH_TYPE = RemoteStoreEnums.PathType.FIXED.getCode();
 
     private final String name;
     private final String id;
+    private final int shardPathType;
     private final int hashCode;
 
+    // Used for testing only
     public IndexId(final String name, final String id) {
+        this(name, id, DEFAULT_SHARD_PATH_TYPE);
+    }
+
+    public IndexId(String name, String id, int shardPathType) {
         this.name = name;
         this.id = id;
+        this.shardPathType = shardPathType;
         this.hashCode = computeHashCode();
-
     }
 
     public IndexId(final StreamInput in) throws IOException {
         this.name = in.readString();
         this.id = in.readString();
+        if (in.getVersion().onOrAfter(Version.CURRENT)) {
```

Comment:
We need the exact version in this case 2.17?

Reply:
This needs to be handled in the followup PR to allow the BWC tests to not fail.

Reply:
No you can do it in one shot, set this to 2.17 to start with